src/datatypes.ts

```diff
@@ -193,71 +187,120 @@ export function serializeTime(value: Time): string {
+          : value.timezoneMinuteOffset);
+    } else {
+      timezoneString = timezoneString + ":00";
+    }
+  }
+
   return (
-    (utcHours < 10 ? "0" + utcHours : utcHours) +
+    (value.hour < 10 ? "0" + value.hour : value.hour) +
     ":" +
-    (utcMinutes < 10 ? "0" + utcMinutes : utcMinutes) +
+    (value.minute < 10 ? "0" + value.minute : value.minute) +
     ":" +
     (value.second < 10 ? "0" + value.second : value.second) +
-    (value.millisecond ? "." + millisecondString + "Z" : "Z")
+    (value.millisecond ? "." + millisecondString : "") +
+    (timezoneString ? timezoneString : "")
   );
 }
 
 /**
  * @internal
  * @param literalString Value to deserialise.
  * @returns Deserialized time, or null if the given value is not a valid serialised datetime.
- * @see https://www.w3.org/TR/xmlschema-2/#time-lexical-representation
+ * @see https://www.w3.org/TR/xmlschema-2/#time-lexical-repr
  */
 export function deserializeTime(literalString: string): Time | null {
   // Time in the format described at
-  // https://www.w3.org/TR/xmlschema-2/#time-lexical-representation
-  // (without constraints on the value).
+  // https://www.w3.org/TR/xmlschema-2/#time-lexical-repr
   // \d\d:\d\d:\d\d - Two digits for the hour, minute and second, respectively, separated by a `:`.
   //                  Example: "13:37:42".
   // (\.\d+)? - Optionally a `.` followed by one or more digits representing milliseconds.
   //            Example: ".1337".
   // (Z|(\+|-)\d\d:\d\d) - The letter Z indicating UTC, or a `+` or `-` followed by two digits for
   //                       the hour offset and two for the minute offset, separated by a `:`.
   //                       Example: "+13:37".
-  const datetimeRegEx = /\d\d:\d\d:\d\d(\.\d+)?(Z|(\+|-)\d\d:\d\d)/;
+  const datetimeRegEx = /\d\d:\d\d:\d\d(\.\d+)?(Z|(\+|-)\d\d:\d\d)?/;
   if (!datetimeRegEx.test(literalString)) {
     return null;
   }
   const [timeString, timezoneString] = splitTimeFromTimezone(literalString);
   const [hourOffset, minuteOffset] = getTimezoneOffsets(timezoneString);
-  if (minuteOffset > 59) {
+  if (minuteOffset > 59 || hourOffset > 24) {
     return null;
   }
   const [hourString, minuteString, timeRest] = timeString.split(":");
-  let utcHours = Number.parseInt(hourString, 10) + hourOffset;
-  let utcMinutes = Number.parseInt(minuteString, 10) + minuteOffset;
+  let utcHours = Number.parseInt(hourString, 10);
+  let utcMinutes = Number.parseInt(minuteString, 10);
   const [secondString, optionalMillisecondString] = timeRest.split(".");
   const utcSeconds = Number.parseInt(secondString, 10);
   const utcMilliseconds = optionalMillisecondString
     ? Number.parseInt(optionalMillisecondString, 10)
     : undefined;
 
-  if (utcMinutes < 0) {
-    utcHours = utcHours - 1;
-    utcMinutes = 60 + utcMinutes;
-  } else if (utcMinutes >= 60) {
+  if (utcMinutes >= 60) {
     utcHours = utcHours + 1;
     utcMinutes = utcMinutes - 60;
```

Comment:
Edge case, but what if `utcMinutes > 120`?

```suggestion
    utcHours = utcHours + Math.floor(utcMinutes / 60);
    utcMinutes = utcMinutes % 60;
```

Reply:
Ah right, that can't happen because there's just two characters for the minutes :)